src/test/mir-opt/const_allocation/rustc.main.ConstProp.after.mir

```diff
@@ -0,0 +1,61 @@
+// MIR for `main`
+// source = MirSource { instance: Item(DefId(0:4 ~ const_allocation[317d]::main[0])), promoted: None }
+// pass_name = ConstProp
+// disambiguator = after
+
+fn  main() -> () {
+    let mut _0: ();                      // return place in scope 0 at $DIR/const_allocation.rs:5:11: 5:11
+    let _1: &[(std::option::Option<i32>, &[&str])]; // in scope 0 at $DIR/const_allocation.rs:6:5: 6:8
+    let mut _2: &&[(std::option::Option<i32>, &[&str])]; // in scope 0 at $DIR/const_allocation.rs:6:5: 6:8
+
+    bb0: {
+        StorageLive(_1);                 // bb0[0]: scope 0 at $DIR/const_allocation.rs:6:5: 6:8
+        StorageLive(_2);                 // bb0[1]: scope 0 at $DIR/const_allocation.rs:6:5: 6:8
+        _2 = const Scalar(alloc0+0): &&[(std::option::Option<i32>, &[&str])]; // bb0[2]: scope 0 at $DIR/const_allocation.rs:6:5: 6:8
+                                         // ty::Const
+                                         // + ty: &&[(std::option::Option<i32>, &[&str])]
+                                         // + val: Value(Scalar(alloc0+0))
+                                         // mir::Constant
+                                         // + span: $DIR/const_allocation.rs:6:5: 6:8
+                                         // + literal: Const { ty: &&[(std::option::Option<i32>, &[&str])], val: Value(Scalar(alloc0+0)) }
+        _1 = (*_2);                      // bb0[3]: scope 0 at $DIR/const_allocation.rs:6:5: 6:8
+        StorageDead(_2);                 // bb0[4]: scope 0 at $DIR/const_allocation.rs:6:8: 6:9
+        StorageDead(_1);                 // bb0[5]: scope 0 at $DIR/const_allocation.rs:6:8: 6:9
+        _0 = ();                         // bb0[6]: scope 0 at $DIR/const_allocation.rs:5:11: 7:2
+        return;                          // bb0[7]: scope 0 at $DIR/const_allocation.rs:7:2: 7:2
+    }
+}
+
+alloc0: DefId(0:3 ~ const_allocation[317d]::FOO[0]): 16 bytes, alignment 8
+00 00 00 00 00 00 00 00 03 00 00 00 00 00 00 00 
+└────────(17)─────────┘ 
+
+alloc17: 72 bytes, alignment 8
+00 00 00 00 __ __ __ __ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 __ __ __ __ 00 00 00 00 00 00 00 00 02 00 00 00 00 00 00 00 01 00 00 00 2a 00 00 00 00 00 00 00 00 00 00 00 03 00 00 00 00 00 00 00 
+                        └─────────(4)─────────┘                                                 └─────────(8)─────────┘                                                 └────────(13)─────────┘ 
+
```

Comment:
It is impossible to see which relocations belong where. This could be fixed by adding line-wrapping, though that is non-trivial in case the wrap is inside a relocation. Or we could find some way to have relocations more "inline"... the current format is somewhat misleading:
```
00 00 00 00 __ __ __ __ 00 00 00 00 00 00 00 00
                        └─────────(4)─────────┘
```
This looks like there's just zeroes in memory, but the last 8 bytes are not actually zero.

Also, instead of just "4", maybe say "alloc4" to make it clear that this is referring to an allocation? Right now these look like random integers.

Reply:
> Or we could find some way to have relocations more "inline"...

Just some quick math:
* on 32-bit, 4 bytes are 11 characters, same as `alloc99+999`
* on 64-bit, 8 bytes are 23 characters, same as `alloc99999999+999999999`

The latter is definitely enough to handle most realistic situations, I'm guessing.

> Also, instead of just "4", maybe say "alloc4" to make it clear that this is referring to an allocation? Right now these look like random integers.

I agree, although to be fair and make the same comparison as before, these would be the widest possible:
* on 32-bit, `└(alloc99)┘`
* on 64-bit, `└(alloc99999999999999)┘`

To make 32-bit relocations more reasonable we could use `@` instead of `alloc`, as silly as that looks.